src/storage/core.rs

```diff
@@ -137,11 +138,21 @@ impl Storage {
     }
 
     /// # Description
-    /// Writes bytes `value` to active blob
-    /// If active blob reaches it limit, create new and close old
-    /// Returns number of bytes, written to blob
+    /// Writes `data` to active blob asyncronously. If active blob reaches it limit, creates new
+    /// and closes old.
     /// # Examples
-    pub async fn write(self, record: Record) -> Result<()> {
+    /// ```no-run
+    /// block_on(async {
+    ///     let key = 42u64.to_be_bytes().to_vec();
+    ///     let data = b"async written to blob".to_vec();
+    ///     await!(storage.write(key, data))
+    /// )};
+    /// ```
+    pub async fn write(self, key: Vec<u8>, data: Vec<u8>) -> Result<()> {
+        if key.len() as u64 != self.shared.config.key_size.ok_or(Error::Unitialized)? {
+            return Err(Error::KeySizeMismatch(key.len()));
```

Comment:
Checking for key size can be put in function. May be it make sense to create some trait for key and not to pass `Vec` every where and we can implement for some basic types (`u64` for example)